src/clib/xtg/cube_export_segy.c

```diff
@@ -212,6 +212,12 @@ cube_export_segy(char *sfile,
             nc = 0;
 
             ilc = x_ijk2ic(i, j, 1, nx, ny, 1, 0);
+            if (ilc < 0) {
+                free(xv);
+                free(yv);
+                throw_exception("Outside boundary");
+                return EXIT_FAILURE;
+            }
```

Comment:
Are we not using `segyio` for segy file handling?

Reply:
The default for export is `xtgeo`, but we could potentially change that to `segyio`. That is a larger change though, @jcrivenaes what is you opinion?

Reply:
The reason was (and is?) that `segyio` is (or was) not good at creating stuff "from nothing"; `segyio` takes an existing segy file, change something and then output it again. When we make a cube from "scratch" in xtgeo, I found it difficult to use `segyio`. That said, it can be worth to see if this is simpler now in never versions of `segyio`.

Reply:
The simple example in segyio was updated 2019. It might not contain all necessary features though, but looks fairly straight forward: https://github.com/equinor/segyio/blob/master/python/examples/make-file.py

Reply:
Looks ok, but I think that is outside the scope of this PR (which is already quite large), as it is seemingly not directly straight forward. So I created an issue for that: https://github.com/equinor/xtgeo/issues/555

Reply:
Thanks for the segyio example. I agree we make an issue on this this (letting segyio do the export) but not as a high pri as the current exports works.

Reply:
Sounds good!